api/v1alpha1/nodepool_types.go

```diff
@@ -55,152 +55,281 @@ type NodePool struct {
 	// Spec is the desired behavior of the NodePool.
 	Spec NodePoolSpec `json:"spec,omitempty"`
 
-	// Status is the most recently observed status of the NodePool.
+	// Status is the latest observed status of the NodePool.
 	Status NodePoolStatus `json:"status,omitempty"`
 }
 
-// NodePoolSpec defines the desired state of NodePool
+// NodePoolSpec is the desired behavior of a NodePool.
 type NodePoolSpec struct {
-	// ClusterName is the name of the Cluster this object belongs to.
+	// ClusterName is the name of the HostedCluster this NodePool belongs to.
+	//
+	// TODO(dan): Should this be a LocalObjectReference?
+	//
 	// +immutable
 	ClusterName string `json:"clusterName"`
+
+	// Release specifies the OCP release used for the NodePool. This informs the
+	// ignition configuration for machines, as well as other platform specific
+	// machine properties (e.g. an AMI on the AWS platform).
+	//
+	// +kubebuilder:validation:Required
+	// +required
```

Comment:
I think `+required` is actually a thing so it can be dropped.

Reply:
Also everything is required unless specified otherwise, so we should probably either set `+kubebuilder:validation:Required` everywhere consistently or just drop it completely. Probably a TODO for a follow up PR if we don't want to do it here.

Reply:
Good point, I removed all the required tags and indeed the CRDs remain the same, so that's a nice cleanup